### PR TITLE
Update kubernetes guide

### LINKED ISSUE #17629: improve kubernetes guide to explain how we use var name including dot
## Description

Proposition: Improve the kubernetes guide to explain that we use var name including dot(s) as currently there is no mention about that.

By example, you can use such a syntax with 2 vars `greeting.message`, `greeting.name` if they are enclosed with double quotes
```
quarkus.kubernetes.env.vars."greeting.message"=Hello from Kubernetes to
quarkus.kubernetes.env.vars."greeting.name"=Quarkus developers
```

to generate such kubernetes Deployment MANIFEST
```
    spec:
      containers:
        - env:
            - name: KUBERNETES_NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
            - name: GREETING_MESSAGE
              value: Hello from Kubernetes to
            - name: GREETING_NAME
              value: Quarkus developers
```

### PR DESCRIPTION
**Pull request by metacosm**
Also fixes #17629.

### CODE REVIEW

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 1]
```diff
@@ -1263,3 +1006,14 @@ while the other files provide the necessary connection information.
...
+[#configuration-reference]
+== Configuration reference
+
+TIP: Remember that it's possible to quote property names when they contain `.`.
+Indeed, without quoting, Quarkus wouldn't be able to tell where your property ends when parsing the property's "path".
+So if you need to reference a
+`greeting.message` property in one of the configuration files, you can use the following syntax:
+`quarkus.whatever."greeting.message".something=Hello`.
+
+include::{generated-dir}/config/quarkus-kubernetes.adoc[opts=optional,leveloffset=+1]
```
Comment (gsmet) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:1019]: Do we have the `quarkus.openshift` and `quarkus.knative` here?

Comment (metacosm) [reply docs/src/main/asciidoc/deploying-to-kubernetes.adoc:1019]: As far as I can tell, yes.

**Comment by geoand** [comment]
The documentation job seems to have failed

**Review by iocanel — CHANGES_REQUESTED** [review]
The tables are removed from `deploying-to-kubernetes.adoc` but not really added to `config-refence.adoc`. 

Most likely these get generated automatically, still I don't see them in https://quarkus.io/guides/config-reference.

So, we might need to do something additional?

**Comment by metacosm** [comment]
> The documentation job seems to have failed

I think this occurs because it seems that when you include a config section using an include directive then it automatically adds a section on Duration which thus gets duplicated and you get the warning/error about duplicated section id.

I know next to nothing about asciidoc or how it's used in Quarkus so not sure what can be done here.

**Comment by metacosm** [comment]
> The tables are removed from `deploying-to-kubernetes.adoc` but not really added to `config-refence.adoc`.
> 
> Most likely these get generated automatically, still I don't see them in https://quarkus.io/guides/config-reference.

If the tables are automatically generated, I don't know how and it's not documented, at least that I could find. Considering they were outdated, I doubt that they are automatically generated (at least, not as a part of the regular build).

It seemed to me that the properties were more adequately described in the configuration reference than in the tables that occurred in the `deploying-to-kubernetes` document, hence why I removed them and tried to replace them with the appropriate section of the reference.

**Comment by geoand** [comment]
What's the status of this?

**Comment by metacosm** [comment]
> What's the status of this?

Not sure :)
I'm not sure how some parts of the documentation are generated, nor do I understand the documentation infrastructure completely so I might be missing parts. I unfortunately don't have time to look into it deeper at this time. Maybe I should extract the commits dealing with adding information and not bother with trying to clean up outdated tables in this PR?

**Comment by geoand** [comment]
Sure yeah

**Comment by gsmet** [comment]
I will have a quick look tomorrow.

**Comment by geoand** [comment]
Closing this as out of date

**Pull request closed without merging**